rpki-validator-app/src/main/scala/net/ripe/rpki/validator/rtr/Pdu.scala

```diff
@@ -285,6 +285,7 @@ object Pdus {
   }
 
   private def parseErrorPdu(buffer: ChannelBuffer, headerShort: Short): Right[Nothing, ErrorPdu] = {
+    // NB: this will break for lengths between 2^31 and 2^32
```

Comment:
Shouldn't it be readUnsignedInt then?

Reply:
This is a tricky bit. readUnsignedInt returns a long, with max value of 2^32-1. But you can't create pretty much anything in Java with such length, the max length of byte array is Integer.MAX_LENGTH, you can't even pass Long value as array's size.
I guess the proper solution is to maintain a list of arrays if the size is bigger than 2^31 :(